bundles/sirix-distributed/src/main/java/org/sirix/wikipedia/hadoop/XMLRecordReader.java

```diff
@@ -0,0 +1,371 @@
+			}
+		} catch (final XMLStreamException e) {
+			LOGWRAPPER.error(e.getMessage(), e);
+		}
+	}
+
+	@Override
+	public DateWritable getCurrentKey() {
+		return mKey;
+	}
+
+	@Override
+	public Text getCurrentValue() {
+		return mValue;
+	}
+
+	@Override
+	public float getProgress() {
+		return mCountEvents;
+	}
+
+	@Override
+	public synchronized void close() throws IOException {
+		try {
+			mReader.close();
+		} catch (final XMLStreamException e) {
+			LOGWRAPPER.error(e.getMessage(), e);
+		}
+	}
+
+	@Override
+	public boolean nextKeyValue() throws IOException, InterruptedException {
+		mValue.clear();
+		boolean retVal = false;
+
+		try {
+			// Skip whitespace.
+			skipWhitespace();
+
+			if (mReader.hasNext() && mReader.peek().isStartElement()
+					&& mReader.peek().asStartElement().getName().equals(mPage)) {
+				mPageEvents = new ArrayList<>();
+				while (mReader.hasNext() && !mBeginFilter.accept(mReader.peek())) {
+					mPageEvents.add(mReader.nextEvent());
+				}
+			}
+
+			for (final XMLEvent event : mPageEvents) {
+				mEventWriter.add(event);
+			}
+
+			// Moves to start of record.
+			final boolean foundStartEvent = moveToEvent(mReader, mBeginFilter, false);
+
+			if (foundStartEvent) {
+				final boolean foundEndEvent = moveToEvent(mReader, mEndFilter, true);
+
+				if (foundEndEvent) {
+					// Add last element to the writer.
+					mEventWriter.add(mReader.nextEvent());
+					skipWhitespace();
+					if (mReader.hasNext() && mReader.peek().isEndElement()
+							&& mReader.peek().asEndElement().getName().equals(mPage)) {
+						mEventWriter.add(mReader.nextEvent());
+					} else {
+						mEventWriter.add(mEventFactory.createEndElement(mPage, null));
+					}
+					retVal = true;
+
+					mWriter.flush();
+					mValue.set(mWriter.toString());
+					mWriter.getBuffer().setLength(0);
+				}
+			}
+		} catch (final XMLStreamException e) {
+			LOGWRAPPER.error(e.getMessage(), e);
+		}
+
+		return retVal;
+	}
+
+	/**
+	 * Skip whitespace events.
+	 * 
+	 * @throws XMLStreamException
+	 *           if any parser error occurs
+	 */
+	private void skipWhitespace() throws XMLStreamException {
+		while (mReader.hasNext() && mReader.peek().isCharacters()
+				&& mReader.peek().asCharacters().isWhiteSpace()) {
+			mReader.next();
+		}
+	}
+
+	/**
+	 * Move to beginning of record.
+	 * 
+	 * @param paramReader
+	 *          XML Reader {@link XMLEventReader}.
+	 * @param paramFilter
+	 *          XML filter {@link EventFilter}.
+	 * @param paramIsRecord
+	 *          determines if the parser is inside a record or outside
+	 * @return false if event was not found and received end of file
+	 * @throws XMLStreamException
+	 *           if a parsing error occurs
+	 */
+	private boolean moveToEvent(final XMLEventReader paramReader,
+			final EventFilter paramFilter, final boolean paramIsRecord)
+			throws XMLStreamException {
+		boolean isTimestamp = false;
+		final DateFormat formatter = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss",
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The SimpleDateFormat object was created without setting the time zone. Make sure that you want to use the default time zone or use letter z, Z or X in the pattern. Otherwise, if you do not call setTimeZone() on the created SimpleDateFormat object, you might get an unexpected date and time when using the object.